File: app/schemas/community.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field
from .user import UserBase

class CommunityBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=1000)
    is_private: bool = False

class CommunityCreate(CommunityBase):
    pass

class CommunityUpdate(CommunityBase):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = None
    is_private: Optional[bool] = None

class CommunityInDB(CommunityBase):
    id: int
    slug: str
    created_at: datetime
    created_by_id: int

    class Config:
        from_attributes = True

class Community(CommunityInDB):
    members_count: int = 0
    created_by: Optional["UserBase"] = None
    is_member: bool = False
    is_admin: bool = False

    class Config:
        from_attributes = True

class CommunityWithMembers(Community):
    members: List["UserBase"] = []

    class Config:
        from_attributes = True

# For handling member operations
class CommunityMemberAdd(BaseModel):
    user_id: int

class CommunityMemberRemove(BaseModel):
    user_id: int

# For response models that need minimal community info
class CommunityMinimal(BaseModel):
    id: int
    name: str
    slug: str

    class Config:
        from_attributes = True

File: frontend/src/types/community.ts
export interface Community {
  id: number;
  name: string;
  slug: string;
  description: string;
  is_private: boolean;
  created_at: string;
  created_by_id: number;
  members_count: number;
  is_member: boolean;
  is_admin: boolean;
}

export interface CommunityCreate {
  name: string;
  description?: string;
  is_private: boolean;
}

export interface CommunityUpdate {
  name?: string;
  description?: string;
  is_private?: boolean;
}

export interface Post {
  id: number;
  title: string;
  content: string;
  author_name?: string;
  created_at: string;
  updated_at: string;
  author_id: number;
  community_id?: number;
  community?: CommunityMinimal;
  likes_count: number;
  comments_count: number;
  is_announcement?: boolean;
  is_liked?: boolean;
  comments?: any[];
}

export interface CommunityMinimal {
  id: number;
  name: string;
  slug: string;
  description: string;
}
